bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ValueCreator.java

```diff
@@ -46,6 +46,15 @@
 public abstract class ValueCreator {
 
     private static final Map<String, ValueCreator> runtimeValueCreators = new HashMap<>();
+    private static Strand mainStrand = null;
+
+    public static void setMainStrand(Strand strand) {
```

Comment:
Shall we move this to scheduler with name daemon strand and set init or stop strand based on case.